test/ci.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of vsc-install,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-install
+#
+# vsc-install is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-install is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-install. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Test CI functionality
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import logging
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+try:
+    from cStringIO import StringIO  # Python 2
+except ImportError:
+    from io import StringIO  # Python 3
+
+import vsc.install.ci
+from vsc.install.ci import JENKINSFILE_REVISION, TOX_INI_REVISION, gen_jenkinsfile, gen_tox_ini, write_file
+from vsc.install.testing import TestCase
+
+
+def read_file(path):
+    """Read file at specified path, and return contents."""
+    with open(path, 'r') as handle:
+        return handle.read()
+
+
+class CITest(TestCase):
+    """License related tests"""
+
+    def setUp(self):
+        """Test setup"""
+        super(CITest, self).setUp()
+
+        self.cwd = os.getcwd()
+        self.tmpdir = tempfile.mkdtemp()
+        self.tmpdir_name = os.path.basename(self.tmpdir)
+
+    def tearDown(self):
+        """Test cleanup"""
+        shutil.rmtree(self.tmpdir)
+        os.chdir(self.cwd)
+
+        super(CITest, self).tearDown()
+
+    def run_function(self, function, *args, **kwargs):
+        """Run specified function with specified arguments, and capture generated stdout/stderr."""
+        orig_handlers = vsc.install.ci.LOG.handlers[:]
+        stringio = StringIO()
+        handler = logging.StreamHandler(stringio)
+        vsc.install.ci.LOG.handlers = [handler]
+
+        function(*args, **kwargs)
+
+        stdout = stringio.getvalue()
+        vsc.install.ci.LOG.handlers = orig_handlers
+
+        return stdout
+
+    def test_gen_jenkinsfile(self):
+        """Test generating of Jenkinsfile."""
```

Comment:
please make this simpler.

Reply:
Please clarify what you mean. ;)

I guess moving the `write_file` to `main` will help quite a bit already...

Reply:
it's a lot of code to compare 1 static piece of text with a generated one.

Reply:
should be a lot better now...